there appears to be a compilation error arising from the test:

* https://build.osrfoundation.org/job/gazebo-ci-pr_any-ubuntu_auto-amd64-gpu-nvidia/1169/console
* https://build.osrfoundation.org/job/gazebo-ci-pr_any-homebrew-amd64/2941/console

~~~
test/integration/transport.cc:164:45: error: no matching member function for call to 'Subscribe'
...
test/integration/transport.cc:179:26: error: no matching member function for call to 'Subscribe'
...
~~~

--- a/gazebo/transport/Node.hh
+++ b/gazebo/transport/Node.hh
@@ -288,7 +288,7 @@ namespace gazebo
 
         return result;
       }
-      
+
       /// \brief Subscribe to a topic using a bost::function as the callback,
       /// allowing to pass lambdas.
       /// \param[in] _topic The topic to subscribe to

--- a/test/integration/transport.cc
+++ b/test/integration/transport.cc
@@ -154,35 +154,32 @@ TEST_F(TransportTest, PubSub)
 }
 
 // Standard pub/sub using lambdas
-TEST_F(TransportTest, PubSubLambda)
+TEST_F(TransportTest, PubSubNoncapturingLambda)
 {
   Load("worlds/empty.world");
 
   transport::NodePtr node = transport::NodePtr(new transport::Node());
   node->Init();
   transport::PublisherPtr scenePub = node->Advertise<msgs::Scene>("~/scene");
-  transport::SubscriberPtr sceneSub = node->Subscribe("~/scene",
-      [](ConstScenePtr &/*_msg*/) -> void{g_sceneMsg=true;});
+  transport::SubscriberPtr sceneSub = node->Subscribe<msgs::Scene>("~/scene",
+      +[](ConstScenePtr & _msg) -> void {
+      	g_sceneMsg=true;
+	  }
+  );
+}
 
-  msgs::Scene msg;
-  msgs::Init(msg, "test");
-  msg.set_name("default");
+TEST_F(TransportTest, PubSubCapturingLambda)
+{
+  Load("worlds/empty.world");
 
-  scenePub->Publish(msg);
-
-  std::vector<transport::PublisherPtr> pubs;
-  std::vector<transport::SubscriberPtr> subs;
-
-  for (unsigned int i = 0; i < 10; ++i)
-  {
-    pubs.push_back(node->Advertise<msgs::Scene>("~/scene"));
-    subs.push_back(node->Subscribe<msgs::Scene>("~/scene", 
-    [](ConstScenePtr &/*_msg*/) -> void {g_sceneMsg=true;}));
-    scenePub->Publish(msg);
-  }
-
-  pubs.clear();
-  subs.clear();
+  transport::NodePtr node = transport::NodePtr(new transport::Node());
+  node->Init();
+  transport::PublisherPtr scenePub = node->Advertise<msgs::Scene>("~/scene");
+  transport::SubscriberPtr sceneSub = node->Subscribe<msgs::Scene>("~/scene",
+      [this](ConstScenePtr & _msg) -> void {
+      	g_sceneMsg=true;
+	  }
+  );
 }
 
 /////////////////////////////////////////////////

--- a/test/integration/transport.cc
+++ b/test/integration/transport.cc
@@ -176,8 +176,8 @@ TEST_F(TransportTest, PubSubLambda)
   for (unsigned int i = 0; i < 10; ++i)
   {
     pubs.push_back(node->Advertise<msgs::Scene>("~/scene"));
-    subs.push_back(node->Subscribe("~/scene", 
-    [](ConstScenePtr &/*_msg*/) -> void{g_sceneMsg=true;}));
+    subs.push_back(node->Subscribe<msgs::Scene>("~/scene", 
+    [](ConstScenePtr &/*_msg*/) -> void {g_sceneMsg=true;}));
     scenePub->Publish(msg);
   }
 

